source/type/data.cpp

```diff
@@ -101,22 +104,45 @@ bool Data::load(const std::string& filename,
         LOG4CPLUS_INFO(logger, boost::format("stopTimes : %d nb foot path : %d Nombre de stop points : %d")
                        % pt_data->nb_stop_times()
                        % pt_data->stop_point_connections.size()
-                       % pt_data->stop_points.size()
-            );
-        if (chaos_database) {
-            fill_disruption_from_database(*chaos_database, *pt_data, *meta, contributors);
-        }
-        build_raptor(raptor_cache_size);
+                       % pt_data->stop_points.size());
     } catch(const wrong_version& ex) {
         LOG4CPLUS_ERROR(logger, "Cannot load data: " << ex.what());
         last_load = false;
+        return last_load;
     } catch(const std::exception& ex) {
         LOG4CPLUS_ERROR(logger, "Data loading failed: " << ex.what());
         last_load = false;
+        return last_load;
     } catch(...) {
         LOG4CPLUS_ERROR(logger, "Data loading failed");
         last_load = false;
+        return last_load;
 	}
+
+    // Load Disruption (optionnal)
+    if (chaos_database) {
+        try {
+            fill_disruption_from_database(*chaos_database, *pt_data, *meta, contributors);
+        } catch(const std::exception& ex) {
```

Comment:
there you should probably reload data (will take a bit of time, but better with clean base_schedule data)
maybe you can be smart : if it fails before doing anything (connection crashing at start) you can avoid reload.

Reply:
I agree!

Reply:
@pbougue if you're concerned by the time-consuming reload, we can reload the data(or not) depend on the raised exception type

Reply:
I'm agree. I'll do in my next commit.